app/components/autocomplete/at_mention/at_mention.js

```diff
@@ -50,74 +50,74 @@ export default class AtMention extends Component {
     constructor(props) {
         super(props);
 
-        const ds = new ListView.DataSource({
-            sectionHeaderHasChanged: (s1, s2) => s1 !== s2,
-            rowHasChanged: (r1, r2) => r1 !== r2
-        });
-        const data = {};
         this.state = {
-            active: false,
-            dataSource: ds.cloneWithRowsAndSections(data)
+            sections: []
         };
     }
 
     componentWillReceiveProps(nextProps) {
-        const {isSearch} = nextProps;
-        const regex = isSearch ? FROM_REGEX : AT_MENTION_REGEX;
-        const match = nextProps.postDraft.substring(0, nextProps.cursorPosition).match(regex);
+        const {autocompleteUsers, hasMatch, isSearch, matchTerm, requestStatus} = nextProps;
 
-        if (!match || this.state.mentionComplete) {
+        if (!hasMatch || this.state.mentionComplete) {
             this.setState({
-                active: false,
-                matchTerm: null,
-                mentionComplete: false
+                mentionComplete: false,
+                sections: []
             });
             return;
         }
 
-        const matchTerm = isSearch ? match[1] : match[2];
-        if (matchTerm !== this.state.matchTerm) {
-            this.setState({
-                matchTerm
-            });
-
+        if (matchTerm !== this.props.matchTerm && requestStatus !== RequestStatus.STARTED) {
             const {currentTeamId, currentChannelId} = this.props;
             this.props.actions.autocompleteUsers(matchTerm, currentTeamId, currentChannelId);
         }
 
-        if (nextProps.requestStatus !== RequestStatus.STARTED) {
-            const membersInChannel = this.filter(nextProps.autocompleteUsers.inChannel, matchTerm) || [];
-            const membersOutOfChannel = this.filter(nextProps.autocompleteUsers.outChannel, matchTerm) || [];
-
-            let data = {};
+        if (requestStatus === RequestStatus.NOT_STARTED || requestStatus === RequestStatus.SUCCESS) {
```

Comment:
This is updating the sections every time any prop changes. Can we look at the individual props that go into making up `sections` to see if any of them have changed?